authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
+      "system:serviceaccount:openshift-cluster-csi-drivers:aws-ebs-csi-driver-controller-sa"
+    ]
+  }
+}
+----
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-machine-api-aws-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-machine-api:machine-api-controllers"
+    ]
+  }
+}
+----
+
+. Copy the manifests that the `sts-preflight` tool created to the `manifests` directory in the same location as the `install-config.yaml` file:
++
+[source,terminal]
+----
+$ cp _output/manifests/* <path-to-install-config.yaml-dir>/manifests/
+----
+
+. Copy the private key for the `ServiceAccount` object signer to the same location as the `install-config.yaml` file:
++
+[source,terminal]
+----
+$ cp -a _output/tls <path-to-install-config.yaml-dir>
+----
+
+. Create a token that is trusted by the OIDC identity provider:
++
+[source,terminal]
+----
+$ ./sts-preflight token
+----
++
+This token is used to assume a role to get the credentials that are required to start the installation. The token is created in the `_output/token` directory.
+
+. Use the token to assume the installer role and mint the credentials that are required for the {product-title} installation process.
++
+[source,terminal]
+----
+$ ./sts-preflight assume
+----
++
+The output prints several `export` commands.
+
+. Run the `export` commands from the output of the previous step.
++
+[source,terminal]
+----
+$ export AWS_ACCESS_KEY_ID=<aws-access-key-id>
+----
++
+[source,terminal]
+----
+$ export AWS_SECRET_ACCESS_KEY=<aws-secret-access_>
+----
++
+[source,terminal]
+----
+$ export AWS_SESSION_TOKEN=<aws-session-token>
+----
+
+[id="sts-mode-installing-preflight-run-installer"]
+==== Running the installer
+
+. Run the {product-title} installer
++
+[source,terminal]
+----
+$ ./openshift-install create cluster --log-level=debug
+----
+
+. Continue to xref:#sts-mode-installing-verifying[Verifying the installation].
+
+[id="sts-mode-installing-manual-config"]
+=== Configuring a cluster manually
+
+When xref:#sts-mode-installing[installing] an {product-title} cluster that is configured to use the CCO in manual mode with STS, you can manually create the required AWS resources.
+
+As an alternative, you can use the xref:#sts-mode-installing-preflight-config[`sts-preflight` tool] to create AWS resources.
+
+.Procedure
+
+. Generate a private key to sign the `ServiceAccount` object:
++
+[source,terminal]
+----
+$ openssl genrsa -out sa-signer 4096
+----
+
+. Generate a `ServiceAccount` object public key:
++
+[source,terminal]
+----
+$ openssl rsa -in sa-signer -pubout -out sa-signer.pub
+----
+
+. Create an S3 bucket to hold the OIDC configuration:
++
+[source,terminal]
+----
+$ aws s3api create-bucket --bucket <oidc-bucket-name> --region <aws-region>
```

Comment:
My ignorance: Can the bucket be named anything or is there a specific name for the bucket already?

Reply:
It looks like [Amazon has some rules for this](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html), but I am guessing that this would be out of scope for our docs (and also I suspect someone doing this process is familiar with said rules)

Reply:
Yes, plenty of rules around naming of buckets. This step should error in obvious ways if you pick an invalid bucket name.